src/core.ts

```diff
@@ -1287,6 +1287,27 @@ export const toBase64 = (str: string | null | undefined): string => {
   throw new OpenAIError('Cannot generate b64 string; Expected `Buffer` or `btoa` to be defined');
 };
 
+/**
+ * Converts a Base64 encoded string to a Float32Array.
+ * @param base64Str - The Base64 encoded string.
+ * @returns An Array of numbers interpreted as Float32 values.
+ */
+export const toFloat32Array = (base64Str: string): Array<number> => {
+  if (typeof Buffer !== 'undefined') {
+    // for Node.js environment
+    return Array.from(new Float32Array(Buffer.from(base64Str, 'base64').buffer));
```

Comment:
curious if you've benchmarked how much of a difference just returning the `Float32Array` directly would have?

if it's a big difference we should probably have an opt-in flag to just do that. (doesn't block this PR)

Reply:
This implementation seems to produce wrong results, see #1448

Reply:
Awesome! Thank you @RIscRIpt for looking into this.